lib/pact_broker/client/cli/pact_commands.rb

```diff
@@ -116,6 +118,17 @@ def branch
                 end
               end
 
+              def consumer_app_version
+                require 'pact_broker/client/git'
+                if options.consumer_app_version.blank? && options.auto_detect_version_properties
+                  PactBroker::Client::Git.commit(raise_error: explict_auto_detect_version_properties)
```

Comment:
Ok, all this "auto detect vs explicit auto detect" logic was because I started by auto detecting the branch, and I was going to make it auto detect the branch by default, but have it *not* raise errors if it couldn't find the branch env var if it was the "default auto detect", but for it to raise errors if the user had explicitly said "auto detect", because I wanted to turn it on without breaking anyone's builds. I'm not sure if that logic needs to be applied to the consumer version, because the consumer version is a required property, but the branch wasn't.

Reply:
> "auto detect vs explicit auto detect"
> 
> Logic was because I started by auto detecting the branch, and 
>  - I was going to make it auto detect the branch by default
>    - don't raise errors if it couldn't find the branch env var if it was the "default auto detect"
>    - raise errors if the user had explicitly said "auto detect",
>    
> I wanted to turn it on without breaking anyone's builds. 
> 
> I'm not sure if that logic needs to be applied to the consumer version, because the consumer version is a required property, but the branch wasn't.

Edited to read it a bit better, I had a look last eve, and I think I am confused. I'll push up some changes

Reply:
Just make this `(raise_error: true)`

Reply:
Still relevant ^^

Reply:
sorry I did pick this up this morning to try and wrap it up but got pulled away on various threads. thanks for the extra reviews comments :)